StRoot/StFttPointMaker/StFttPointMaker.cxx

```diff
@@ -0,0 +1,214 @@
+{ 
+    return kStOk;
+}
+
+//_____________________________________________________________
+Int_t
+StFttPointMaker::FinishRun( Int_t runnumber )
+{ 
+    return kStOk;
+}
+
+//_____________________________________________________________
+Int_t
+StFttPointMaker::Finish()
+{ 
+    return kStOk;
+}
+
+
+//_____________________________________________________________
+Int_t
+StFttPointMaker::Make()
+{ 
+    LOG_INFO << "StFttPointMaker::Make()" << endm;
+
+    mEvent = (StEvent*)GetInputDS("StEvent");
+    if(mEvent) {
+        LOG_DEBUG<<"Found StEvent"<<endm;
+    } else {
+        return kStOk;
+    }
+    mFttCollection=mEvent->fttCollection();
+    if(!mFttCollection) {
+        return kStOk;
+    }
+
+    mFttDb = static_cast<StFttDb*>(GetDataSet("fttDb"));
+
+    assert( mFttDb );
+
+    if ( mUseTestData )
+        InjectTestData();
+
+    MakeLocalPoints();
+    MakeGlobalPoints();
+
+    LOG_INFO << "StFttPointMaker made " << mFttCollection->numberOfPoints() << " points this event" << endm;
+
+    return kStOk;
+}
+
+void StFttPointMaker::InjectTestData(){
+    mFttCollection->rawHits().clear();
+
+    // TODO: inject clean strip hits to test cluster finder
+    // StFttRawHit *hit = new StFttRawHit( sec, rdo, feb, vm, vmm[0].ch, vmm[0].adc, vmm[0].bcid, vmm[0].tb );
+    // hit->setMapping( plane, quadrant, row, strip ) 
+}
+
+void StFttPointMaker::MakeLocalPoints(){
+    // next we will need them in even more detail
+    // per strip group, but start here
+    // key, dir, value is cluster
+    // std::map< UChar_t, std::vector<StFttCluster *> > clustersPerRob[4][StFttDb::nRowsPerQuad];
+    std::vector< StFttCluster *> clusters[StFttDb::nRob][StFttDb::nRowsPerQuad][StFttDb::nStripOrientations];
+
+    for ( StFttCluster* clu : mFttCollection->clusters() ) {
+        UChar_t rob = mFttDb->rob( clu );
+        if ( clu->nStrips() < 2 ) continue;
+        clusters[ rob ][ clu->row() ][ clu->orientation() ].push_back( clu );
+    } // loop on hit
+
+    
+    for ( size_t iRob = 1; iRob < StFttDb::nRob; iRob ++ ){
+        for ( size_t iRowH = 0; iRowH < 3; iRowH++ ){
+            size_t nH = clusters[ iRob ][ iRowH ][ kFttHorizontal ].size();
+            for ( size_t iRowV = 0; iRowV < 3; iRowV++ ){
+                size_t nV = clusters[ iRob ][ iRowV ][ kFttVertical ].size();
+                for ( size_t iH = 0; iH < nH; iH++ ){
+                    auto cluH = clusters[ iRob ][ iRowH ][ kFttHorizontal ][ iH ];
+                    for ( size_t iV = 0; iV < nV; iV++ ){
+                        auto cluV = clusters[ iRob ][ iRowV ][ kFttVertical ][ iV ];
+
+                        StFttPoint * p = makePoint( cluH, cluV );
+                    } // iV
+                } // iH
+            } // iRowV
+        } // iRowH
+    } // iRob
+} // MakeLocalPoints
+
+void StFttPointMaker::MakeGlobalPoints() {
+    for ( StFttPoint * p : mFttCollection->points() ){
+
+        double z = p->plane() * 60;
```

Comment:
Global transformation is hardcoded... what is the plan to apply alignment measurements?

Reply:
sTGC is a bit behind on the DB side, so we have no actual databases setup yet. But your point is taken, we can at least setup the infrastructure to query the StFttDbMaker for these values instead of hard coding.

Reply:
Addressed: 3686a6eea3b300c95180bdcee0ad698b5763a88c